File: AgentieTurismJava/src/main/java/org/example/agentietursimjava/Service/Service.java
package org.example.agentietursimjava.Service;

import org.example.agentietursimjava.Domain.Angajat;
import org.example.agentietursimjava.Domain.Excursie;
import org.example.agentietursimjava.Domain.Rezervare;

import java.util.Optional;
import java.util.UUID;

public class Service {
    private final AngajatService angajatService;
    private final ExcursieService excursieService;
    private final RezervareService rezervareService;

    public Service(AngajatService angajatService, ExcursieService excursieService, RezervareService rezervareService) {
        this.angajatService = angajatService;
        this.excursieService = excursieService;
        this.rezervareService = rezervareService;
    }

    // Angajat
    public Optional<Angajat> findAngajat(UUID uuid) {
        return angajatService.findAngajat(uuid);
    }

    public Iterable<Angajat> findAllAngajati() {
        return angajatService.findAllAngajati();
    }

    public Optional<Angajat> addAngajat(Angajat angajat) {
        return angajatService.addAngajat(angajat);
    }

    public Optional<Angajat> deleteAngajat(Angajat angajat) {
        return angajatService.deleteAngajat(angajat);
    }

    public Optional<Angajat> updateAngajat(Angajat angajat) {
        return angajatService.updateAngajat(angajat);
    }

    public Optional<Angajat> findAngajatByUsername(String username) {
        return angajatService.findAngajatByUsername(username);
    }

    // Excursie
    public Optional<Excursie> findExcursie(UUID uuid) {
        return excursieService.findExcursie(uuid);
    }

    public Iterable<Excursie> findAllExcursii() {
        return excursieService.findAllExcursii();
    }

    public Optional<Excursie> addExcursie(Excursie excursie) {
        return excursieService.addExcursie(excursie);
    }

    public Optional<Excursie> deleteExcursie(Excursie excursie) {
        return excursieService.deleteExcursie(excursie);
    }

    public Optional<Excursie> updateExcursie(Excursie excursie) {
        return excursieService.updateExcursie(excursie);
    }

    public Iterable<Excursie> cautaExcursii(String numeObiectiv, Integer deLaOra, Integer panaLaOra) {
        return excursieService.cautaExcursii(numeObiectiv, deLaOra, panaLaOra);
    }

    // Rezervare
    public Optional<Rezervare> findRezervare(UUID uuid) {
        return rezervareService.findRezervare(uuid);
    }

    public Iterable<Rezervare> findAllRezervari() {
        return rezervareService.findAllRezervari();
    }

    public Optional<Rezervare> addRezervare(Rezervare rezervare) {
        return rezervareService.addRezervare(rezervare);
    }

    public Optional<Rezervare> deleteRezervare(Rezervare rezervare) {
        return rezervareService.deleteRezervare(rezervare);
    }

    public Optional<Rezervare> updateRezervare(Rezervare rezervare) {
        return rezervareService.updateRezervare(rezervare);
    }
}

File: AgentieTursimCS/Service/Service.cs
#nullable enable
using System;
using System.Collections.Generic;
using AgentieTurismCS.Domain;

namespace AgentieTurismCS.Service
{
    public class Service
    {
        private readonly AngajatService _angajatService;
        private readonly ExcursieService _excursieService;
        private readonly RezervareService _rezervareService;

        public Service(AngajatService angajatService, ExcursieService excursieService, RezervareService rezervareService)
        {
            _angajatService = angajatService;
            _excursieService = excursieService;
            _rezervareService = rezervareService;
        }
        
        // Angajat
        public IEnumerable<Angajat> FindAllAngajati()
        {
            return _angajatService.FindAllAngajati();
        }
        
        public Angajat? FindAngajatByUsername(string username)
        {
            return _angajatService.FindAngajatByUsername(username);
        }
        
        public Angajat? FindAngajat(Guid guid)
        {
            return _angajatService.FindAngajat(guid);
        }
        
        public Angajat? AddAngajat(Angajat angajat)
        {
            return _angajatService.AddAngajat(angajat);
        }
        
        public Angajat? UpdateAngajat(Angajat angajat)
        {
            return _angajatService.UpdateAngajat(angajat);
        }
        
        public Angajat? DeleteAngajat(Angajat angajat)
        {
            return _angajatService.DeleteAngajat(angajat);
        }
        
        // Excursie
        public IEnumerable<Excursie> FindAllExcursii()
        { 
            return _excursieService.FindAllExcursii();
        }
        
        public Excursie? FindExcursie(Guid guid)
        {
            return _excursieService.FindExcursie(guid);
        }
        
        public Excursie? AddExcursie(Excursie excursie)
        {
            return _excursieService.AddExcursie(excursie);
        }
        
        public Excursie? UpdateExcursie(Excursie excursie)
        {
            return _excursieService.UpdateExcursie(excursie);
        }
        
        public Excursie? DeleteExcursie(Excursie excursie)
        {
            return _excursieService.DeleteExcursie(excursie);
        }
        
        public IEnumerable<Excursie> CautaExcursii(string obiectivTuristic, int deLaOra, int panaLaOra)
        {
            return _excursieService.CautaExcursii(obiectivTuristic, deLaOra, panaLaOra);
        }
        
        // Rezervare
        public IEnumerable<Rezervare> FindAllRezervari()
        {
            return _rezervareService.FindAllRezervari();
        }
        
        public Rezervare? FindRezervare(Guid guid)
        {
            return _rezervareService.FindRezervare(guid);
        }
        
        public Rezervare? AddRezervare(Rezervare rezervare)
        {
            return _rezervareService.AddRezervare(rezervare);
        }
        
        public Rezervare? UpdateRezervare(Rezervare rezervare)
        {
            return _rezervareService.UpdateRezervare(rezervare);
        }
        
        public Rezervare? DeleteRezervare(Rezervare rezervare)
        {
            return _rezervareService.DeleteRezervare(rezervare);
        }
    }
}